action/protocol/staking/protocol.go

```diff
@@ -55,11 +54,10 @@ var (
 
 // Protocol defines the protocol of handling staking
 type Protocol struct {
-	addr        address.Address
-	candCenters *cache.ThreadSafeLruCache
-	depositGas  DepositGas
-	sr          protocol.StateReader
-	config      Configuration
+	addr       address.Address
+	depositGas DepositGas
+	sr         protocol.StateReader
```

Comment:
why we have this sr stored in protocol?

Reply:
yes, this is also a lingering issue. b/c we still have ActiveCandidates(ctx context.Context, height uint64), we need to read the view from sr

ActiveCandidates is for the oldest governance protocol, so comes with a height that you can read the cand list (from election service) at that Ether height

we may need to clearly re/define ActiveCandidates(), for staking v2 it should always be current height, and we need to pass in StateReader, so we can remove p.sr from protocol